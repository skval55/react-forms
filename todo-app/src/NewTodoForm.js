import React, { useState } from "react";

const NewTodoForm = ({ addTask }) => {
  const INITIAL_STATE = { todo: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = (e) => {
    e.preventDefault();
    addTask(formData);
    setFormData(INITIAL_STATE);
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="todo">Todo:</label>
      <input
        id="todo"
        name="todo"
        value={formData.todo}
        onChange={handleChange}
      />
      <button>Add</button>
    </form>
  );
};

export default NewTodoForm;
