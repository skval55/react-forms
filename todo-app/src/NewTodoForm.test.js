import { render, screen } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// smoke test
test("renders without crashing", () => {
  render(<NewTodoForm addTask={() => console.log("test")} />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <NewTodoForm addTask={() => console.log("test")} />
  );
  expect(asFragment()).toMatchSnapshot();
});
