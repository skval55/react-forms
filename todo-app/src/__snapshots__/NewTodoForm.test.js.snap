// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo:
    </label>
    <input
      id="todo"
      name="todo"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
